nomad/job_endpoint_test.go

```diff
@@ -6858,87 +6873,103 @@ func TestJobEndpoint_Dispatch(t *testing.T) {
 
 			var dispatchResp structs.JobDispatchResponse
 			dispatchErr := msgpackrpc.CallWithCodec(codec, "Job.Dispatch", tc.dispatchReq, &dispatchResp)
+			if dispatchErr != nil {
+				must.True(t, tc.err)
```

Comment:
I realize this is existing code, but this is asserting the test case field. I'd probably refactor this test a little bit to combine `tc.err` and `tc.errStr` into a `tc.expectError` field (of type `error`), and then do something along the lines of:

```go
if tc.expectError != nil {
	must.ErrorContains(t, dispatchErr, tc.expectError.Error())
} else {
	must.NoError(t, dispatchErr)
}
```

I think you'll find that pattern gets used in a lot of our tests.